File: challenge1/bmi.c
#include <stdio.h>
#include <string.h>
#define MAX_LENGTH 20

float bmi(float weight, float height){
    float f;
    f = weight/(height*height);
    return f;
}

char * id(int age, char gender, char* name, char*country) {
    char id[26];
    strncat(id,"%s", country);
    strncat(id,"%s",name);
    return id;
}
                                                                 //TODO
int main(void) {
    
    char name[MAX_LENGTH];
    int age;
    char country[MAX_LENGTH];
    char gender;
    float height;
    float weight;
    
	printf("What's your name?\n");
	scanf("%s",name);
	printf("\nWhere are you from?\n");
	scanf("%s",country);
	printf("\nHow old are you?\n");
	scanf("%d",&age);
	printf("\nWhat's your gender?(M/F)\n");
	scanf("%c",&gender);
	printf("\nHow much do you weigh?\n");
	scanf("%f",&weight);
	printf("\nHow tall are you?\n");
	scanf("%f",&height);
	printf("\n %f \n",bmi(weight,height));
	printf("\n %s \n", id(age, gender, name, country));
	return 0;
}

File: challenge1/bmi.java
import java.util.Scanner;
import java.util.Map;
import java.util.HashMap;
import java.lang.String;

/**
 * @author Maslor
 */
 
public class BMI {
	
	public Map<String,String> countrytags = new HashMap<String,String>();
	
	public BMI() {
	countrytags.put("PORTUGAL","PT");
	countrytags.put("FRANCE","FR");
	countrytags.put("SPAIN","ES");
	countrytags.put("ITALY","IT");
	countrytags.put("UNITED KINGDOM","UK");
	countrytags.put("GERMANY","DE");
	}
	
	Scanner sc = new Scanner(System.in);

	public int readInt(){
		int i=0;
		i = sc.nextInt();
		return i;
	}

	public float readFloat(){
		float in;
		in = sc.nextFloat();
		return in;
	}

	public String readString(){
		String in=null;
		in = sc.next();
		return in;
	}

	public String getCountryTag(String country) {
		return countrytags.get(country);
	}

	public float bmi(float weight, float height) {
		return (weight / (height * height));
	}

	public String id(String name, String country, int age, String gender) {
		return "" + getCountryTag(country.toUpperCase()) + age + gender.toUpperCase() + name.toUpperCase();
	}

	public void finalMessage(String id, String name, float bmi) {
		System.out.printf("Hello %s\nYour code is: %s\nYour bmi is: %s\n",name,id,bmi);
	}
	
		
		public static void main(String[] args) {
			BMI bmi = new BMI();
			System.out.println("What's your name?");
			String name = bmi.readString();
			System.out.println("How old are you?");
			int age = bmi.readInt();
			System.out.println("Where are you from?");
			String country = bmi.readString();
			System.out.println("What's your gender?(m/f)");
			String gender = bmi.readString();
			System.out.println("How tall are you?");
			float height = bmi.readFloat();
			System.out.println("How much do you weigh?(kilos)");
			float weight = bmi.readFloat();
			
			bmi.finalMessage(bmi.id(name,country,age,gender), name, bmi.bmi(weight, height));
			
		}
	}



